lib/core/base/audit.js

```diff
@@ -144,7 +144,9 @@ Audit.prototype.run = function (context, options, resolve, reject) {
 	'use strict';
 	this.validateOptions(options);
 
-	axe._tree = axe.utils.getFlattenedTree(document.documentElement); //cache the flattened tree
+	if (!axe._tree) {
```

Comment:
Why conditional? We always want to do this to ensure that the state of the flattened tree is accurate

Reply:
Well, because I didn't want to overhaul this thing so close to the release... The right way to do this would've been to have Context keep the tree and not store it on axe._tree. It than could've made sure it was based on elements that have the same ownerDocument, and flatten it instead of whatever is on the global document.... That's just a lot of work. :/ I'll see what I can do.